website/src/auth.tsx

```diff
@@ -1,53 +1,95 @@
 import { HubCallback } from "@aws-amplify/core"
+import {
+  AuthenticationDetails,
+  CognitoUser,
+  CognitoUserPool,
+  CognitoUserSession,
+} from "amazon-cognito-identity-js"
 import Amplify, { Auth, Hub } from "aws-amplify"
-import React, { createContext, useContext, useState } from "react"
+import React, { createContext, useContext, useEffect, useState } from "react"
 
-const UserContext = createContext<CognitoUser | null>(null)
+type UserContextType = {
+  user: CognitoUser | null
+  setUser: (user: CognitoUser | null) => void
+  operations: {
+    loginUser: (username: string, password: string) => void
+  }
+}
+
+const UserContext = createContext<UserContextType>({} as UserContextType)
+
+export const userPool = new CognitoUserPool({
+  UserPoolId: process.env["DAILP_USER_POOL"] ?? "",
+  ClientId: process.env["DAILP_USER_POOL_CLIENT"] ?? "",
+})
 
 export const UserProvider = (props: { children: any }) => {
-  const [user, setUser] = useState<CognitoUser | null>(null)
-  React.useEffect(() => {
-    Amplify.configure({
-      Auth: {
-        // TODO How to get these variables from the server?
-        region: process.env["DAILP_AWS_REGION"],
-        userPoolId: process.env["DAILP_USER_POOL"],
-        userPoolWebClientId: process.env["DAILP_USER_POOL_CLIENT"],
-      },
-    })
+  const [user, setUser] = useState<CognitoUser | null>(
+    userPool.getCurrentUser()
+  )
 
-    Auth.currentUserPoolUser()
-      .then((user) => setUser(user))
-      .catch(() => setUser(null))
-
-    const listener: HubCallback = async (data) => {
-      switch (data.payload.event) {
-        case "signIn":
-        case "signUp":
-          setUser(await Auth.currentUserPoolUser())
-          break
-        case "signOut":
-          setUser(null)
-          break
-      }
+  // refreshes last authenticated user
+  useEffect(() => {
+    if (user != null) {
+      user.getSession(function (err: Error, result: CognitoUserSession | null) {
```

Comment:
I'm confused, why are you calling this function then doing nothing with the result? Just to log when you're automatically signed in?

Reply:
While CognitoPool.getCurrentUser() gets the last authenticated user from the userpool, if the page is refreshed, the localStorage is cleared along with the user's tokens, so I added a UseEffect on component mount that will refresh the user's tokens (and restore them to the local storage). The refresh of the tokens is done automatically when the CognitoUser.getSession() method is called.